### PR TITLE
Deal with the 2.5 deprecations

### PR DESCRIPTION
**Pull request by alex**

### CODE REVIEW

File: tests/hazmat/primitives/test_x25519.py [hunk 1]
```diff
@@ -95,16 +94,11 @@ def test_null_shared_key_raises_error(self, backend):
...
-        key = X25519PrivateKey.generate().public_key()
-        with pytest.warns(utils.DeprecatedIn25):
-            key.public_bytes()
-
     def test_public_bytes_bad_args(self, backend):
         key = X25519PrivateKey.generate().public_key()
         with pytest.raises(ValueError):
             key.public_bytes(None, serialization.PublicFormat.Raw)
-        with pytest.raises(ValueError):
+        with pytest.raises(TypeError):
```
Comment (reaperhulk) [inline tests/hazmat/primitives/test_x25519.py]: Is this test useful now?

File: tests/hazmat/primitives/test_ec.py [hunk 2]
```diff
@@ -181,7 +181,7 @@ def test_encode_point():
         16
     )
     pn = ec.EllipticCurvePublicNumbers(x, y, ec.SECP256R1())
-    with pytest.warns(utils.DeprecatedIn25):
+    with pytest.warns(utils.CryptographyDeprecationWarning):
```
Comment (reaperhulk) [inline tests/hazmat/primitives/test_ec.py]: Why not use PersistentlyDeprecatedIn2019 here?

File: tests/hazmat/primitives/test_x25519.py (hunk 1 above)
Comment (alex) [reply tests/hazmat/primitives/test_x25519.py]: 🤷‍♂ it doesn't hurt.

File: tests/hazmat/primitives/test_ec.py (hunk 2 above)
Comment (alex) [reply tests/hazmat/primitives/test_ec.py]: Uhhh, wtf, I have no idea.

**Pull request merged**